Extension/src/LanguageServer/settings.ts

```diff
@@ -996,28 +849,90 @@ export class OtherSettings {
-    public get filesAutoSaveAfterDelay(): boolean { return vscode.workspace.getConfiguration("files").get("autoSave") === "afterDelay"; }
-    public get editorInlayHintsEnabled(): boolean { return vscode.workspace.getConfiguration("editor.inlayHints").get<string>("enabled") !== "off"; }
-    public get editorParameterHintsEnabled(): boolean | undefined { return vscode.workspace.getConfiguration("editor.parameterHints").get<boolean>("enabled"); }
-    public get searchExclude(): vscode.WorkspaceConfiguration | undefined { return vscode.workspace.getConfiguration("search", this.resource).get("exclude"); }
-    public get workbenchSettingsEditor(): string | undefined { return vscode.workspace.getConfiguration("workbench.settings").get<string>("editor"); }
 
-    public get colorTheme(): string | undefined { return vscode.workspace.getConfiguration("workbench").get<string>("colorTheme"); }
+    private getVSCodeSettingAsNumber(setting: string, configuration: string, resource: any, defaultNumber?: number): number | undefined {
+        const fullConfiguration = vscode.workspace.getConfiguration(setting, resource);
+        const value = fullConfiguration.get<number>(configuration);
+        if (isNumber(value)) {
+            return value;
+        }
+        const config = fullConfiguration.inspect<number>(configuration);
+        return config?.defaultValue ?? defaultNumber;
+    }
 
-    public getCustomColorToken(colorTokenName: string): string | undefined { return vscode.workspace.getConfiguration("editor.tokenColorCustomizations").get<string>(colorTokenName); }
-    public getCustomThemeSpecificColorToken(themeName: string, colorTokenName: string): string | undefined { return vscode.workspace.getConfiguration(`editor.tokenColorCustomizations.[${themeName}]`, this.resource).get<string>(colorTokenName); }
+    private getVSCodeSettingAsKeyValueObject(setting: string, configuration: string, keyType: string, valueType: string, resource?: any): any {
+        const fullConfiguration = vscode.workspace.getConfiguration(setting, resource);
+        const value = fullConfiguration.get<number>(configuration);
+        if (isValidMapping(value, keyType, valueType)) {
+            return value;
+        }
+        const config = fullConfiguration.inspect<number>(configuration);
+        return config?.defaultValue;
+    }
 
+    private isArrayOfTextMateRules(x: any): x is TextMateRule[] {
+        return isArray(x) && x.every(y => this.isTextMateRule(y));
+    }
+
+    private isTextMateRule(x: any): x is TextMateRule {
+        return this.isTextMateRuleScope(x.scope) && this.isTextMateRuleSettings(x.settings);
+    }
+
+    private isTextMateRuleScope(x: any): boolean {
+        // Assuming scope can be either a string or an array of strings
+        return isString(x) || (isArray(x) && x.every(isString));
+    }
```

Comment:
we have `isArrayOfString` that you should use here instead of reimplementing it.

I actually don't know what the scope is meant to be because the type is declared as `any` in our code. If we know the type should be `string | string[]`, the definition of `TextMateRule` should be updated.

Reply:
After looking at the documentation, it looks like it is expecting an array of strings. I will change the syntax accordingly.